### PR TITLE
fix: migrate sync/fetch personal cards txn to platform

### PR DESCRIPTION
**Pull request by sumrender**
## Clickup
https://app.clickup.com/t/86cwm1x52

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

## Summary by CodeRabbit

- **New Features**
	- Enhanced transaction fetching capabilities with support for both public and platform APIs.
	- Added a new method to hide transactions using the platform API.

- **Improvements**
	- Updated transaction fetching methods to accept additional parameters for better flexibility.
	- Improved organization and clarity of test cases for transaction synchronization.
	- Enhanced transaction loading processes by integrating platform-specific logic.

- **Bug Fixes**
	- Refined logic for managing loading states based on transaction counts.

### CODE REVIEW

File: src/app/core/services/personal-cards.service.ts [hunk 1]
```diff
@@ -350,7 +350,22 @@ export class PersonalCardsService {
     return this.getBankTransactions(params, usePlatformApi).pipe(map((res) => res.count));
   }
 
-  fetchTransactions(accountId: string): Observable<ApiV2Response<PersonalCardTxn>> {
+  fetchTransactionsPlatform(accountId: string): Observable<PlatformApiResponse<{}>> {
```
Comment (Chethan-Fyle) [inline src/app/core/services/personal-cards.service.ts:353]: Don't use `{}`. Define the model/interface and use it.

File: src/app/core/services/personal-cards.service.ts [hunk 2]
```diff
@@ -350,7 +350,22 @@ export class PersonalCardsService {
...
+        personal_card_id: accountId,
+      },
+    };
+    return this.spenderPlatformV1ApiService.post('/personal_card_transactions', payload);
+  }
+
+  fetchTransactions(
+    accountId: string,
+    usePlatformApi: boolean
+  ): Observable<ApiV2Response<PersonalCardTxn> | PlatformApiResponse<{}>> {
```
[the remark this answers is not in the document]
Comment (Chethan-Fyle) [reply src/app/core/services/personal-cards.service.ts:365]: +1

**Review by Chethan-Fyle — CHANGES_REQUESTED** [review]
One important comment.

File: src/app/core/models/platform/platform-personal-card-syn-txns.model.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface PersonalCardSyncTxns {}
```
Comment (Chethan-Fyle) [inline src/app/core/models/platform/platform-personal-card-syn-txns.model.ts:2]: It says `Txns` (that is multiple transactions) but it is not an array. Make it `PersonalCardSyncTxn` and use array of it (that is `PersonalCardSyncTxn[]`) during usage.

Also, why is it empty?

**Review by Chethan-Fyle — CHANGES_REQUESTED** [review]
One comment and a doubt.

File: src/app/core/models/platform/platform-personal-card-syn-txns.model.ts (hunk 3 above)
Comment (sumrender) [reply src/app/core/models/platform/platform-personal-card-syn-txns.model.ts:2]: @Chethan-Fyle sir, Receiving an empty object response only for this endpoint.

The api role is `Sync Personal Card Transactions`. So, I've named the interface to be PersonalCardSyncTxns

**Review by Chethan-Fyle — APPROVED** [review]

**Pull request merged**